src/data_structures/src/merkle_tree.cairo

```diff
@@ -54,11 +76,30 @@ impl MerkleTreeImpl of MerkleTreeTrait {
                     // Compute the hash of the current node and the current element of the proof.
                     // We need to check if the current node is smaller than the current element of the proof.
                     // If it is, we need to swap the order of the hash.
-                    if Into::<felt252, u256>::into(current_node) < (*proof_element).into() {
-                        current_node = LegacyHash::hash(current_node, *proof_element);
-                    } else {
-                        current_node = LegacyHash::hash(*proof_element, current_node);
-                    }
+                    current_node = match self.hash_method {
+                        HashMethod::Pedersen(()) => {
```

Comment:
Finds this kinda odd to have a match on the the HashMethod, because then if a new hashing gets added it'll break stuff

What is the issue with making it a shared var?

Reply:
Do you suggest to have the state in the struct and do a `self.state.sate = 0` before each hash computation?

Reply:
I think there will be a type issue, since HashState is not a generic Type, I cannot take it as it is, and I'll have to make a struct for each hash method.
Also I don't see why it would break something if a new hash method is added, since it must just add a new enum option and add corresponding matches